kernel/src/main/java/org/kframework/unparser/ToLatex.java

```diff
@@ -0,0 +1,208 @@
+ */
+public class ToLatex {
+
+    public static byte[] apply(K k) {
+        try {
+            ByteArrayOutputStream out = new ByteArrayOutputStream();
+            apply(new DataOutputStream(out), k);
+            return out.toByteArray();
+        } catch (IOException e) {
+            throw KEMException.criticalError("Could not write K term to LaTeX", e, k);
+        }
+    }
+
+    public static String latexedKLabel(String orig) {
+        return "klabel" + orig.replaceAll("[^a-zA-Z]", "");
+    }
+
+    private static void writeString(DataOutputStream out, String str) throws IOException {
+        out.write(str.getBytes(StandardCharsets.UTF_8));
+    }
+
+    public static void apply(DataOutputStream out, Att att) throws IOException {
+        writeString(out, ("\\outerAtt{" + att.toString() + "}"));
+    }
+
+    public static byte[] apply(Definition def) {
+        try {
+            ByteArrayOutputStream out = new ByteArrayOutputStream();
+            apply(new DataOutputStream(out), def);
+            return out.toByteArray();
+        } catch (IOException e) {
+            throw KEMException.criticalError("Could not write K Definition to LaTeX", e);
+        }
+    }
+
+    public static void apply(DataOutputStream out, Definition def) throws IOException {
+        writeString(out, ("\\outerDefinition{"));
+        for (Module mod: JavaConverters.seqAsJavaList(def.modules().toSeq())) {
+            apply(out, mod);
+        }
+        writeString(out, "}{");
+        apply(out, def.att());
+        writeString(out, "}");
+    }
+
+    public static byte[] apply(Module mod) {
+        try {
+            ByteArrayOutputStream out = new ByteArrayOutputStream();
+            apply(new DataOutputStream(out), mod);
+            return out.toByteArray();
+        } catch (IOException e) {
+            throw KEMException.criticalError("Could not write K Module to LaTeX", e);
+        }
+    }
+
+    public static void apply(DataOutputStream out, Module mod) throws IOException {
+        writeString(out, ("\\outerModule{" + mod.name() + "}{"));
+        for (Module m: JavaConverters.seqAsJavaList(mod.imports().toSeq())) {
+            writeString(out, "\\outerImport{" + m.name() + "}");
+        }
+        writeString(out, "}{");
+        for (Sentence sent: JavaConverters.seqAsJavaList(mod.localSentences().toSeq())) {
+            apply(out, sent);
+        }
+        writeString(out, ("}{" + mod.att() + "}"));
+    }
+
+    public static byte[] apply(Sentence sent) {
+        try {
+            ByteArrayOutputStream out = new ByteArrayOutputStream();
+            apply(new DataOutputStream(out), sent);
+            return out.toByteArray();
+        } catch (IOException e) {
+            throw KEMException.criticalError("Could not write K term to LaTeX", e, sent);
+        }
+    }
+
+    public static void apply(DataOutputStream out, Sentence sent) throws IOException {
+        if (sent instanceof Rule) {
```

Comment:
it would be nice if we used polymorphism for this instead of a big if/else block.

Reply:
I'm pulling out the `Sentence` and `Module` serializers from this PR, only including the `K` one for now.